de1plus/profile.tcl

```diff
@@ -520,41 +520,110 @@ namespace eval ::profile {
+            set filepath $filename
+        }
+        
+        if { ![file exists $filepath] } {
+            msg -WARNING [namespace current] find_file: "profile file '$filename' not found"
+            return ""
+        }
+        
+        return $filepath
+    }
+        
+    # Based on proc load_settings_vars, but only loads profile variables, and returns a list with the profile (which is
+    # normally then coerced to an array) instead of loading the profile into the global settings. 
+    # Reads from different sources, extracting only the profile-related vars from the source.
+    proc read_legacy { src_type {src {}} {extra_vars 1} } {
+        if { $src_type eq "next" || $src_type eq "settings" || $src_type eq "array_name" } {
+            if { $src_type eq "next" || $src_type eq "settings" } {
+                upvar "::settings" src_array 
+            } else {
+                upvar $src src_array
+            }
+        } elseif { $src_type eq "shot_file" || $src_type eq "profile_file" } {
+            set filepath [find_file $src 1]
+            if { $filepath eq {} } {
+                return
+            }
+            
+            try {
+                array set src_array [encoding convertfrom utf-8 [read_binary_file $filepath]]
+            } on error err {
+                msg -WARNING [namespace current] read_legacy: "error reading file '$filepath': $err"
+                return
+            }
+        
+            if { $src_type eq "shot_file" } {
+                array set src_array $src_array(settings)
+            }
+        } elseif { $src_type eq "list" } {
+            array set src_array [$src]
+        }
+        
+#        msg -INFO [namespace current] read_profile: "reading file '$profile_path'"
         array set profile {}
     
-        # Set defaults in case they are not defined in the profile file 
+        # Set defaults in case they are not defined in the source data 
         set profile(final_desired_shot_volume_advanced_count_start) 0
         set profile(settings_profile_type) "settings_2a"
         set profile(beverage_type) "espresso"
     
-        try {
-            array set arrfile [encoding convertfrom utf-8 [read_binary_file $profile_path]]
-        } on error err {
-            msg -WARNING [namespace current] read_profile: "error reading profile '$fn': $err"
-            return
+        if { [string is true $extra_vars] } {
+            set profile_vars [concat [::profile_vars] profile_filename profile_to_save original_profile_title]
+        } elseif { $extra_vars eq {} } {
+            set profile_vars [::profile_vars]
+        } else {
+            set profile_vars [concat [::profile_vars] $extra_vars]
         }
-    
-        foreach k [concat [::profile_vars] $extra_vars] {
-            if { [info exists arrfile($k)] } {
-                set profile($k) $arrfile($k)
-            } elseif { ![info exists profile($k)] } {
-                set profile($k) {}
+        
```

Comment:
nit: whitespace

Reply:
Not sure what's the whitespace issue here , is it the blank line 621? I see many lines with that (variable number of whitespace characters), would you like those removed on all blank lines or what?